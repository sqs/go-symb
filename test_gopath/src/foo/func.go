package foo

func A(b, c string, d bool) (e, f int, g uint) {
	bb := b
	d = true
	e = 7
	f = len(c + bb)
	g = uint(e)
	return
}
