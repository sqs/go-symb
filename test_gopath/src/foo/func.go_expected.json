[
  {
    "Expr": "foo",
    "Ident": "foo",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 8,
      "Line": 1,
      "Column": 9
    },
    "ExprType": "",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "",
      "Offset": 0,
      "Line": 0,
      "Column": 0
    },
    "ReferObj": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "Local": false,
    "Universe": false,
    "IsDecl": false
  },
  {
    "Expr": "A",
    "Ident": "A",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 18,
      "Line": 3,
      "Column": 6
    },
    "ExprType": "func(b string, c string, d bool) (e int, f int, g uint)",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 18,
      "Line": 3,
      "Column": 6
    },
    "ReferObj": {
      "Isa": "Func",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "A",
      "Type": "func(b string, c string, d bool) (e int, f int, g uint)"
    },
    "Local": false,
    "Universe": false,
    "IsDecl": true
  },
  {
    "Expr": "b",
    "Ident": "b",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 20,
      "Line": 3,
      "Column": 8
    },
    "ExprType": "string",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 20,
      "Line": 3,
      "Column": 8
    },
    "ReferObj": {
      "Isa": "Var",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "b",
      "Type": "string"
    },
    "Local": true,
    "Universe": false,
    "IsDecl": true
  },
  {
    "Expr": "c",
    "Ident": "c",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 23,
      "Line": 3,
      "Column": 11
    },
    "ExprType": "string",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 23,
      "Line": 3,
      "Column": 11
    },
    "ReferObj": {
      "Isa": "Var",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "c",
      "Type": "string"
    },
    "Local": true,
    "Universe": false,
    "IsDecl": true
  },
  {
    "Expr": "string",
    "Ident": "string",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 25,
      "Line": 3,
      "Column": 13
    },
    "ExprType": "string",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "",
      "Offset": 0,
      "Line": 0,
      "Column": 0
    },
    "ReferObj": {
      "Isa": "TypeName",
      "Pkg": null,
      "Name": "string",
      "Type": "string"
    },
    "Local": false,
    "Universe": true,
    "IsDecl": false
  },
  {
    "Expr": "d",
    "Ident": "d",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 33,
      "Line": 3,
      "Column": 21
    },
    "ExprType": "bool",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 33,
      "Line": 3,
      "Column": 21
    },
    "ReferObj": {
      "Isa": "Var",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "d",
      "Type": "bool"
    },
    "Local": true,
    "Universe": false,
    "IsDecl": true
  },
  {
    "Expr": "bool",
    "Ident": "bool",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 35,
      "Line": 3,
      "Column": 23
    },
    "ExprType": "bool",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "",
      "Offset": 0,
      "Line": 0,
      "Column": 0
    },
    "ReferObj": {
      "Isa": "TypeName",
      "Pkg": null,
      "Name": "bool",
      "Type": "bool"
    },
    "Local": false,
    "Universe": true,
    "IsDecl": false
  },
  {
    "Expr": "e",
    "Ident": "e",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 42,
      "Line": 3,
      "Column": 30
    },
    "ExprType": "int",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 42,
      "Line": 3,
      "Column": 30
    },
    "ReferObj": {
      "Isa": "Var",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "e",
      "Type": "int"
    },
    "Local": true,
    "Universe": false,
    "IsDecl": true
  },
  {
    "Expr": "f",
    "Ident": "f",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 45,
      "Line": 3,
      "Column": 33
    },
    "ExprType": "int",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 45,
      "Line": 3,
      "Column": 33
    },
    "ReferObj": {
      "Isa": "Var",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "f",
      "Type": "int"
    },
    "Local": true,
    "Universe": false,
    "IsDecl": true
  },
  {
    "Expr": "int",
    "Ident": "int",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 47,
      "Line": 3,
      "Column": 35
    },
    "ExprType": "int",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "",
      "Offset": 0,
      "Line": 0,
      "Column": 0
    },
    "ReferObj": {
      "Isa": "TypeName",
      "Pkg": null,
      "Name": "int",
      "Type": "int"
    },
    "Local": false,
    "Universe": true,
    "IsDecl": false
  },
  {
    "Expr": "g",
    "Ident": "g",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 52,
      "Line": 3,
      "Column": 40
    },
    "ExprType": "uint",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 52,
      "Line": 3,
      "Column": 40
    },
    "ReferObj": {
      "Isa": "Var",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "g",
      "Type": "uint"
    },
    "Local": true,
    "Universe": false,
    "IsDecl": true
  },
  {
    "Expr": "uint",
    "Ident": "uint",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 54,
      "Line": 3,
      "Column": 42
    },
    "ExprType": "uint",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "",
      "Offset": 0,
      "Line": 0,
      "Column": 0
    },
    "ReferObj": {
      "Isa": "TypeName",
      "Pkg": null,
      "Name": "uint",
      "Type": "uint"
    },
    "Local": false,
    "Universe": true,
    "IsDecl": false
  },
  {
    "Expr": "bb",
    "Ident": "bb",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 63,
      "Line": 4,
      "Column": 2
    },
    "ExprType": "string",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 63,
      "Line": 4,
      "Column": 2
    },
    "ReferObj": {
      "Isa": "Var",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "bb",
      "Type": "string"
    },
    "Local": true,
    "Universe": false,
    "IsDecl": true
  },
  {
    "Expr": "b",
    "Ident": "b",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 69,
      "Line": 4,
      "Column": 8
    },
    "ExprType": "string",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 20,
      "Line": 3,
      "Column": 8
    },
    "ReferObj": {
      "Isa": "Var",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "b",
      "Type": "string"
    },
    "Local": true,
    "Universe": false,
    "IsDecl": false
  },
  {
    "Expr": "d",
    "Ident": "d",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 72,
      "Line": 5,
      "Column": 2
    },
    "ExprType": "bool",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 33,
      "Line": 3,
      "Column": 21
    },
    "ReferObj": {
      "Isa": "Var",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "d",
      "Type": "bool"
    },
    "Local": true,
    "Universe": false,
    "IsDecl": false
  },
  {
    "Expr": "true",
    "Ident": "true",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 76,
      "Line": 5,
      "Column": 6
    },
    "ExprType": "bool",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "",
      "Offset": 0,
      "Line": 0,
      "Column": 0
    },
    "ReferObj": {
      "Isa": "Const",
      "Pkg": null,
      "Name": "true",
      "Type": "untyped boolean",
      "Val": true
    },
    "Local": false,
    "Universe": true,
    "IsDecl": false
  },
  {
    "Expr": "e",
    "Ident": "e",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 82,
      "Line": 6,
      "Column": 2
    },
    "ExprType": "int",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 42,
      "Line": 3,
      "Column": 30
    },
    "ReferObj": {
      "Isa": "Var",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "e",
      "Type": "int"
    },
    "Local": true,
    "Universe": false,
    "IsDecl": false
  },
  {
    "Expr": "f",
    "Ident": "f",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 89,
      "Line": 7,
      "Column": 2
    },
    "ExprType": "int",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 45,
      "Line": 3,
      "Column": 33
    },
    "ReferObj": {
      "Isa": "Var",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "f",
      "Type": "int"
    },
    "Local": true,
    "Universe": false,
    "IsDecl": false
  },
  {
    "Expr": "len",
    "Ident": "len",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 93,
      "Line": 7,
      "Column": 6
    },
    "ExprType": "\u003ctype of len\u003e",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "",
      "Offset": 0,
      "Line": 0,
      "Column": 0
    },
    "ReferObj": {
      "Isa": "Func",
      "Pkg": null,
      "Name": "len",
      "Type": "\u003ctype of len\u003e"
    },
    "Local": false,
    "Universe": true,
    "IsDecl": false
  },
  {
    "Expr": "c",
    "Ident": "c",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 97,
      "Line": 7,
      "Column": 10
    },
    "ExprType": "string",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 23,
      "Line": 3,
      "Column": 11
    },
    "ReferObj": {
      "Isa": "Var",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "c",
      "Type": "string"
    },
    "Local": true,
    "Universe": false,
    "IsDecl": false
  },
  {
    "Expr": "bb",
    "Ident": "bb",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 101,
      "Line": 7,
      "Column": 14
    },
    "ExprType": "string",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 63,
      "Line": 4,
      "Column": 2
    },
    "ReferObj": {
      "Isa": "Var",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "bb",
      "Type": "string"
    },
    "Local": true,
    "Universe": false,
    "IsDecl": false
  },
  {
    "Expr": "g",
    "Ident": "g",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 106,
      "Line": 8,
      "Column": 2
    },
    "ExprType": "uint",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 52,
      "Line": 3,
      "Column": 40
    },
    "ReferObj": {
      "Isa": "Var",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "g",
      "Type": "uint"
    },
    "Local": true,
    "Universe": false,
    "IsDecl": false
  },
  {
    "Expr": "uint",
    "Ident": "uint",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 110,
      "Line": 8,
      "Column": 6
    },
    "ExprType": "uint",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "",
      "Offset": 0,
      "Line": 0,
      "Column": 0
    },
    "ReferObj": {
      "Isa": "TypeName",
      "Pkg": null,
      "Name": "uint",
      "Type": "uint"
    },
    "Local": false,
    "Universe": true,
    "IsDecl": false
  },
  {
    "Expr": "e",
    "Ident": "e",
    "IdentPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 115,
      "Line": 8,
      "Column": 11
    },
    "ExprType": "int",
    "Pkg": {
      "Isa": "Package",
      "Name": "foo",
      "ImportPath": ""
    },
    "FileName": "foo",
    "ReferPos": {
      "Filename": "test_gopath/src/foo/func.go",
      "Offset": 42,
      "Line": 3,
      "Column": 30
    },
    "ReferObj": {
      "Isa": "Var",
      "Pkg": {
        "Isa": "Package",
        "Name": "foo",
        "ImportPath": ""
      },
      "Name": "e",
      "Type": "int"
    },
    "Local": true,
    "Universe": false,
    "IsDecl": false
  }
]
